### ISSUE TITLE
CouchDB throws 412 when creating a database that already exists, express-pouchdb does not.

### ISSUE THREAD

**Issue by tyler-johnson** [pouchdb/express-pouchdb#217]
I have noticed that CouchDB will throw a 412 Precondition Failed when trying to PUT an existing database. express-pouchdb on the other hand just returns a 201 status on both new and existing databases. This makes it impossible to both create a database and know if it already exists in a single request.

So, is this intentional? And can this be fixed?

**Comment by marten-de-vries** [comment]
We do have [code to handle that use case](https://github.com/pouchdb/express-pouchdb/blob/master/lib/routes/db.js#L17), so I assume you're hitting a bug. Some debugging would probably be necessary to explain what's going on.

I just tried to reproduce your bug using pouchdb-server, but failed:
```bash
marten@procyon:~$ curl -X PUT http://localhost:6984/test
{"ok":true}
marten@procyon:~$ curl -X PUT http://localhost:6984/test
{"error":"file_exists","reason":"The database could not be created."}
```

**Comment by tyler-johnson** [comment]
So it seems like it doesn't work specifically with an encoded "/" or %2F:

```sh
[test] $ curl -X PUT http://localhost:6984/test%2Fother
{"ok":true}
[test] $ curl -X PUT http://localhost:6984/test%2Fother
{"ok":true}
```

**Comment by marten-de-vries** [comment]
Thanks, that made it possible for me to reproduce. Probably some url encode/decode problem when looking up if the db already exists, then.

**Comment by nolanlawson** [comment]
fixed by 7299e100d71cef4ca00fea69282ff53bac24e064
